tests/functional/openstackdataplanedeployment_controller_test.go

```diff
@@ -53,11 +102,77 @@ var _ = Describe("Dataplane Deployment Test", func() {
 		})
 
 		It("should have conditions set", func() {
+
+			nodeSet := dataplanev1.OpenStackDataPlaneNodeSet{}
+			baremetal := baremetalv1.OpenStackBaremetalSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      nodeSet.Name,
+					Namespace: nodeSet.Namespace,
+				},
+			}
+			// Create config map for OVN service
+			ovnConfigMapName := types.NamespacedName{
+				Namespace: namespace,
+				Name:      "ovncontroller-config",
+			}
+			mapData := map[string]interface{}{
+				"ovsdb-config": "test-ovn-config",
+			}
+			th.CreateConfigMap(ovnConfigMapName, mapData)
+
+			// Set node set deployment ready condition to False
+			Eventually(func(g Gomega) {
+				g.Expect(th.K8sClient.Get(th.Ctx, dataplaneNodeSetName, &nodeSet)).To(Succeed())
+				nodeSet.Status.Conditions.MarkFalse(
+					condition.Type(fmt.Sprintf(dataplanev1.NodeSetServiceDeploymentReadyCondition, nodeSet.Name, "download-cache")),
+					condition.RequestedReason,
+					condition.SeverityInfo,
+					fmt.Sprintf(dataplanev1.NodeSetServiceDeploymentReadyMessage, nodeSet.Name, "download-cache"))
+				g.Expect(th.K8sClient.Status().Update(th.Ctx, &nodeSet)).To(Succeed())
+			}, th.Timeout, th.Interval).Should(Succeed())
```

Comment:
You can remove this Eventually block (you should never set Status on a CR that is reconciled by the operator under test). When you remove this you need to add to read back the nodeSet once before you start iterating the services of it to have the defaulted service list in its Spec.
```
			nodeSet = *GetDataplaneNodeSet(dataplaneNodeSetName)
```